### PR TITLE
Add prefect-hex recipe

### PR DESCRIPTION
**Pull request by khuyentran1401**
## Description

Article and video for the prefect-hex collection

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## New Recipe Checklist

- [x] My PR is in the format of `Add <project-name> recipe`
- [x] My recipe is reproducible and explains everything needed to run successfully. If my code has external dependencies, I make mention of them.
- [x] My code is easily understandable and/or well-commented.
- [x] If my recipe requires a new category (e.g. creating a monitoring/ folder in devops/), I have encapsulated my project within its own subfolder so that others can add their own recipes as well.
- [x] If my recipe uses Prefect < 2.0, my code is within the prefect-v1-legacy/ folder.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,12 @@ What are you interested in seeing examples of? [Jot down your big idea here.](ht
 #### Pydantic
 - [Using Pydantic BaseModel with Prefect 2](https://discourse.prefect.io/t/use-pydantic-to-ensure-data-consistency/1815)
 
+#### Hex
+- Create Observable and Reproducible Notebooks with Hex
```
Comment (serinamarie) [inline README.md:133]:
```suggestion
- [Create Observable and Reproducible Notebooks with Hex - Article](https://towardsdatascience.com/create-observable-and-reproducible-notebooks-with-hex-460e75818a09)
- [Create Observable and Reproducible Notebooks with Hex - Video](https://youtu.be/_BjqCrun4nE)
```

**Review by serinamarie — COMMENTED** [review]
I think this looks cleaner. Thoughts?

File: README.md (hunk 1 above)
Comment (khuyentran1401) [reply README.md:133]: That looks good to me.

File: README.md [hunk 2]
```diff
@@ -129,6 +129,13 @@ What are you interested in seeing examples of? [Jot down your big idea here.](ht
 #### Pydantic
 - [Using Pydantic BaseModel with Prefect 2](https://discourse.prefect.io/t/use-pydantic-to-ensure-data-consistency/1815)
 
+#### Hex
+- [Create Observable and Reproducible Notebooks with Hex - Article](https://towardsdatascience.com/create-observable-and-reproducible-notebooks-with-hex-460e75818a09)
+- [Create Observable and Reproducible Notebooks with Hex - Video](https://youtu.be/_BjqCrun4nE)
+  - [Article](https://towardsdatascience.com/create-observable-and-reproducible-notebooks-with-hex-460e75818a09)
+  - [Video](https://youtu.be/_BjqCrun4nE)
+  - [Code](https://github.com/khuyentran1401/customer_segmentation/tree/prefect2)
```
Comment (serinamarie) [inline README.md:137]:
```suggestion
```

**Review by serinamarie — APPROVED** [review]

**Pull request merged**